Update the project overview diagram

coderbyheart opened the pull request:
- [x] wait for support to enable creation of new container packages

The [project overview diagram](https://github.com/NordicSemiconductor/asset-tracker-cloud-docs/blob/f561d88e6835f12074d55418838c29910bbad53c/docs/project/system-overview.jpg) is outdated.

Instead I would like to add an up to date version, that also has links to the respective components. This can be done nicely using Graphviz, with it's SVG export, however including the SVG in the sphinx build needs more investigation: it needs to be embedded as a plain SVG not as an image.

The [source file](./docs/project/system-overview.dot) is used to generate the image:

![image](https://user-images.githubusercontent.com/188915/115230418-9a97f580-a114-11eb-9f53-3a130a675a1b.png)

There is also a variant for the release documentation which only contains the AWS components:

![image](https://user-images.githubusercontent.com/188915/115231902-67eefc80-a116-11eb-9d0f-f663fdaf6573.png)

peknis commented:
Gell Geolocation  -> Cell Geolocation, I think.

peknis commented on docs/project/SystemOverview.rst:
I would leave out the mention about the popularity of JavaScript.

peknis replied:
A couple of lines below, "**It** provides... refers to what exactly? The web application, the cloud, the whole system? Always, be careful when saying "it". The reader needs to know where it refers to.

peknis replied:
Avoid passive voice and talk to the reader who is using these instructions. For example: "After the cloud account has been configured, ...". Could you perhaps say "After you have configured the cloud account, it provides..."

coderbyheart replied:
> I would leave out the mention about the popularity of JavaScript.

Why?

coderbyheart replied:
> Avoid passive voice and talk to the reader who is using these instructions. For example: "After the cloud account has been configured, ...". Could you perhaps say "After you have configured the cloud account, it provides..."

That's not appropriate, because it's not the user that configures the cloud account, it's CDK which provides a template to CloudFormation (which is an infrastructure as a service service by AWS) that does the configuration. All the user has to do is to issue a command and sit back and wait.

I especially want to provide tools that do most of the heavy lifting for the user, so they don't have to do much. That's why I think passive voice here is approporiate.

peknis replied:
The popularity of JavaScript is not relevant here.
Right, I did not get what is done by the system and what the user needs to do. Anyway, we need to make this clear to the reader.

coderbyheart replied:
> The popularity of JavaScript is not relevant here.

It is because it's popularity is one of the reasons why TypeScript was chosen as the programming language for the project.

peknis replied:
Do you always need to justify your choice of the programming language? Believe me, this does not belong in technical customer document.

coderbyheart replied:
> Do you always need to justify your choice of the programming language? 

Yes, because it's not a random choice. Especially for cloud development there are widely used alternatives like Java, Python, or Go and therefore it's good to give users the information why JavaScript is used here, because it affects every line of code in the project.

> Believe me, this does not belong in technical customer document.

Why? What harm does it cause to include this?

peknis replied:
I understand that you want to justify the language choice in a private project, but in a large international company, you are not explaining why you ended up using a certain language. At least not in technical customer documentation.

coderbyheart replied:
Why not?

umapraseeda replied:
I also suggest that we remove this sentence.

coderbyheart replied:
This is not part of the PR.

And, again, why should this be removed? I have asked multiple times for an explanation: **what is so harmful about this explanation that it needs to removed?**

coderbyheart replied:
The explanation is that
> it goes against the principles of not adding additional information that gets in the way of the user flow. I also see this content as extraneous information, it really doesn't help the user to know the reason a choice has been made. So, the explanation is not harmful, but from a stylistic point of view, it is not considered good usability. 

I've **moved** the technical details therefore to an ADR.

coderbyheart committed:
feat: update the project overview diagram (#105)

This turns the project overview diagram in a SVG with
clickable components. The SVG is generated using GraphViz
which allows to maintain the image using text.
This simplifies maintaining links and future additions of
supported cloud providers.

Co-authored-by: Uma Praseeda <uma.praseeda@nordicsemi.no>

umapraseeda commented on docs/project/SystemOverview.rst:
```suggestion
The nRF Asset Tracker provides the tools to configure the developer's cloud account for use with the developer's devices and the single-page application (SPA).
```
